test/core.jl

```diff
@@ -5463,6 +5464,7 @@ t4 = vcat(A23567, t2, t3)
 @test t4[11:15] == A23567
 
 for U in unboxedunions
+    Base.unionlen(U) > 5 && continue # larger values cause subtyping to crash
```

Comment:
Related? If so, how? Does the better inference perform more/other subtype checks that now trigger the crash?